src/Build/BackEnd/Components/SdkResolution/CachingSdkResolverService.cs

```diff
@@ -41,7 +41,7 @@ public override SdkResult ResolveSdk(int submissionId, SdkReference sdk, Logging
 
             bool wasResultCached = true;
 
-            MSBuildEventSource.Log.CachedSdkResolverServiceResolveSdkStart(sdk.Name, solutionPath, projectPath);
+            MSBuildEventSource.Log.CachedSdkResolverServiceResolveSdkStart(sdk.Name, solutionPath ?? string.Empty, projectPath ?? string.Empty);
```

Comment:
I'm guessing exactly one of these is null? Should we verify that before calling this?

Reply:
Why do you think that? I think we could resolve an SDK from an in-memory project (no path) that doesn't have a solution.

Reply:
I can go back and check. I thought in-memory projects have fake paths we use for caching and such. I seem to remember a bug where we neglected to check if a project existed before we tried to open it, but I may be thinking of solution metaprojects specifically.